_posts/2022-mm-dd-crypto-1/2022-mm-dd-crypto1-history-and-basics.markdown

```diff
@@ -0,0 +1,157 @@
+* All examples will assume the standard latin alphabet of A to Z
+* We'll use uppercase letters only
+
+#### Glossary
+There's a few words you'll regularly come across when talking about cryptography. The following list will also only provide a first set of basics and will most probably be expanded as we go along.
+|word         |meaning                                                                                                          |
+|-------------|-----------------------------------------------------------------------------------------------------------------|
+|alphabet     |Set of characters (letters, numbers, symbols, ...) used to form texts. Need not be a conventional alphabet.      |
+|plain text   |Original message, unencrypted and readable by anyone.                                                            |
+|cipher (text)|Encrypted message, unintelligible and (hopefully) safe to pass on.                                               |
+|key (word)   |Phrase or word used to start the encryption algorithm.                                                           |
+|encrypting   |Using a cryptographic algorithm to convert plain text into a cipher.                                             |
+|decrypting   |Using a cryptographic algorithm to recover the original plain text from a cipher.                                |
+|attack       |Attempt to decrypt or modify a cipher text not intended for the attacker.                                        |
+|breaking     |Finding a way of quickly solving a cryptographic problem. May invalidate the security of an entire crypto system.|
+
+Now, with the preamble done, let's get to the matter at hand!
+
+### What are crypto systems?
+Simply speaking, a crypto system is a category of cryptographic methods that follow the same basic principle. Crypto systems themselves usually belong to one of two categories: symmetric or asymmetric. Each crypto system has its own specific strengths and weaknesses and means of being attacked. Before we continue, we will formally define what a crypto system is in mathematical terms. Because as we go on, this is something we'll refer to a lot. So buckle up, this is going to be a bit of a read.
+
+
+#### Definition: crypto system
+Mathematically speaking, a crypto system needs 5 components to be considered complete:
+
+* A finite set of plain texts
+* A finite set of cipher texts
+* A finite set of cryptographic keys
+* A finite set of encryption rules using the above keys
+* A finite set of decryption rules using the above keys
+
+A finite set simply means that the number of elements within the set cannot be infinite (- imagine them as a bag. You can only put in so many things before there's no space left). Usually, the above criteria are represented by symbols grouped into something called a tuple, so we don't have to write the entire list each time. For these posts we'll use the following symbols:
```

Comment:
Key generation is also important to consider. For symmetric cryptography, it's often trivial, but it *needs* to be specified. You can't use a caesar-cipher key (Coming from `Z/nZ`) with AES (which requires keys in the form of `{0,1}^128`) and vice versa.

The sets also don't need to be finite. Look e.g. at any mode of operation for block ciphers, stream ciphers, or even better: Hash functions. They all have an infinite input set for plaintext and ciphertext (although there may be some attacks possible when e.g. a counter repeats, which will happen eventually as they're finite - not a direct problem for hash functions, but pigeonhole principle applies).

```suggestion
Mathematically speaking, a cryptosystem needs 6 components to be considered complete:

* A set of plain texts
* A set of cipher texts
* A set of cryptographic keys
* A key generation function (probabilistic) 
* A encryption function (either deterministic or probabilistic)
* A decryption function (deterministic)

Usually, the above criteria are represented by symbols grouped into something called a tuple, so we don't have to write the entire list each time. For these posts we'll use the following symbols:
```

I think it is also be important to cover whether the functions/algorithms are probabilistic or deterministic. This might become important "later-on", as a deterministic encryption function can never reach indistinguishability under chosen plaintext attacks (IND-CPA), at least as long as it doesn't keep any state.